lib/screens/speaker/speaker_repository.dart

```diff
@@ -0,0 +1,46 @@
+import 'dart:math';
+
+import 'package:ecellapp/core/res/errors.dart';
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+
+@immutable
+abstract class SpeakerRepository {
+  /// All subfunctions are final No arguments required returns json
+  Future<List<dynamic>> speakers();
+}
+
+class FakeSpeakerRepository implements SpeakerRepository {
+  @override
+  Future<List<dynamic>> speakers() async {
```

Comment:
give better names to functions.
```suggestion
  Future<List<dynamic>> getSpeakerList() async {
```